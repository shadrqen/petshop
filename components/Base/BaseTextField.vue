<template>
  <v-text-field
    :id="textId"
    :class="textClass"
    :data-test-id="textDataTestId"
    :label="textLabel"
    :outlined="outlined"
    :rules="textRule"
    :prepend-icon="prependIcon"
    :single-line="singleLine"
    :hide-details="hideDetails"
    :solo="solo"
    @input="$emit('input', $event)"
  />
</template>
<script>
export default {
  name: 'BaseTextField',
  props: {
    textId: {
      type: String,
      required: false,
      default: ''
    },
    textClass: {
      type: String,
      required: false,
      default: ''
    },
    outlined: {
      type: Boolean,
      required: false,
      default: false
    },
    textLabel: {
      type: String,
      required: true
    },
    textDataTestId: {
      type: String,
      required: false,
      default: ''
    },
    prependIcon: {
      type: String,
      required: false,
      default: ''
    },
    singleLine: {
      type: Boolean,
      required: false,
      default: false
    },
    hideDetails: {
      type: Boolean,
      required: false,
      default: false
    },
    solo: {
      type: Boolean,
      required: false,
      default: false
    },
    textRule: {
      type: Array,
      required: false,
      default: () => []
    }
  }
}
</script>

<style scoped>

</style>
