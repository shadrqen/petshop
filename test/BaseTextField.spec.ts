import BaseTextField from '~/components/Base/BaseTextField.vue'
import { createLocalVue, shallowMount } from '@vue/test-utils'

describe('@/components/Base/BaseTextField.vue', () => {
  const localVue = createLocalVue()

  it('should render component and match snapshot', () => {
    const wrapper = shallowMount(BaseTextField, {
      localVue,
      propsData: {
        textLabel: 'email'
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
