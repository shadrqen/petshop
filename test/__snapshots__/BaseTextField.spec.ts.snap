// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@/components/Base/BaseTextField.vue should render component and match snapshot 1`] = `"<v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" appendicon=\\"\\" backgroundcolor=\\"\\" id=\\"\\" label=\\"email\\" prependicon=\\"\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" type=\\"text\\" data-test-id=\\"\\" class=\\"\\"></v-text-field-stub>"`;
